.github/workflows/ci.yaml

```diff
@@ -291,24 +293,21 @@ jobs:
         CIBW_PRERELEASE_PYTHONS: 'True'
         CIBW_TEST_REQUIRES: pytest setuptools  # 3.12+ no longer includes distutils, just always ensure setuptools is present
         CIBW_TEST_COMMAND: PYTHONUNBUFFERED=1 python -m pytest ${{ matrix.test_args || '{project}' }}  # default to test all
-      run: |
-        set -eux
-        
-        mkdir cffi
-        
-        tar zxf ${{ steps.fetch_sdist.outputs.download-path }}/cffi*.tar.gz/cffi*.tar.gz --strip-components=1 -C cffi
-        python -m pip install --upgrade "${{ matrix.cibw_version || 'cibuildwheel' }}"
-
-        # actually build libffi + wheel (using env tweaks above)
-        python -m cibuildwheel --output-dir dist ./cffi
+      with:
+        package-dir: >-
+          ${{ steps.fetch_sdist.outputs.download-path
+          }}/${{ needs.python_sdist.outputs.artifact_name }}
 
-        echo "artifact_name=$(ls ./dist/)" >> "$GITHUB_OUTPUT"
+    - name: determine built wheel filename
+      id: built-artifact-lookup
+      run: echo "artifact_name=$(ls ./wheelhouse/)" >> "${GITHUB_OUTPUT}"
+      shell: bash -eEuxo pipefail {0}
 
     - name: upload artifacts
       uses: actions/upload-artifact@v4
       with:
-        name: ${{ steps.build.outputs.artifact_name }}
-        path: dist/*.whl
+        name: ${{ steps.built-artifact-lookup.outputs.artifact_name }}
+        path: wheelhouse/${{ steps.built-artifact-lookup.outputs.artifact_name }}
```

Comment:
*(I replaced a wildcard with the known value, since we actually have it in a var already)*